### DISCUSSION TITLE
Calculator Button on the Toolbar

### DISCUSSION THREAD

**Discussion by lewinspfeile** [WinEdt-Team/WinEdt-Support#57] [1.2 Bugs and Issues]
I would like to report a minor issue with the calculator button on the toolbar.
The button doesn't seem to work unless I change
`MACRO="Run('calc.exe','%!P',0,0,'Calculator',1,1);"`
to
`MACRO="Run('calc.exe','%!P',0,0,'Calculator');"`
in the Main Menu.ini.
I don't know if the problem can be replicated on other machines.

WinEdt version: 11.2, 64-bit (build 2024613)
OS version: Windows 10, version 22H2, build 19405.4598

**Comment by Alex-Simonic** [comment]
I could not reproduce this on my Windows 11 (and the last time I checked it worked on Windows 10 as well).

But there is always a potential problem with the `"Append=1"` parameter when using the `Run` macro. If the Window with the specified caption (in this case "Calculator") exists then this macro doesn't launch a new instance and attempts to focus the existing Window. But this may be the wrong Window (with the same caption) and the action will fail. This parameter is only used for Command Prompt, Calculator, MiKTeX Console, and TeX Live Shell. If it is removed then a new instance of the application is started. In the case of MiKTeX Console and TeX Live Shell this isn't very pleasant but for Calculator or Command Prompt this may very well be the preferred behavior. If more users experience such problems I may change the default settings for the next version to avoid this situation...
